### PR TITLE
Refactor memberships list

### PR DESCRIPTION
**Pull request by EjPlatzer**
The Involvements component was designed to display a list of a user's memberships, either on the MyProfile view or on the Profile view. It accepted props for the array of memberships to display and whether it was displaying for MyProfile or Profile. It would use the (poorly named) MyProfileActivityList and ProfileActivityList components (which were actually cards for one specific membership).

These have all been deeply refactored, reducing the extensive code duplication and over-engineering. The result is much shorter, much simpler, much easier to read, and much less bug-prone.

### CODE REVIEW

File: src/components/MembershipsList/index.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+
+  useEffect(() => {
+    async function loadMemberships() {
+      setLoading(true);
+      if (myProf) {
+        const myMemberships = await userService.getMembershipsAlphabetically(user);
+        await Promise.all(
+          myMemberships.map(async (membership) => {
+            const involvement = await activity.get(membership.ActivityCode);
+            membership.IsInvolvementPrivate = involvement.Privacy;
```
Comment (joshuaRogers24) [inline src/components/MembershipsList/index.js:31]: Was this occurrence of the term "involvement" difficult to change to membership, or did you think the term actually makes sense here?

File: src/components/MembershipsList/index.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        const publicMemberships = await userService.getPublicMemberships(user);
+        setMemberships(publicMemberships);
+      }
+      setLoading(false);
+    }
+    loadMemberships();
+  }, [myProf, user]);
+
+  const MembershipsList = () => {
+    if (memberships.length === 0) {
```
Comment (joshuaRogers24) [inline src/components/MembershipsList/index.js:45]: Wow, nice simplification here by removing the redundant code that gets public memberships.

File: src/components/MembershipsList/index.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    }
+  };
+
+  const toggleMembershipPrivacy = async (membership) => {
+    try {
+      await membershipService.toggleMembershipPrivacy(membership);
+      createSnackbar(membership.Privacy ? 'Membership Shown' : 'Membership Hidden', 'success');
+      setMemberships(
+        memberships.map((m) => {
+          if (m.MembershipID === membership.MembershipID) m.Privacy = !m.Privacy;
```
Comment (joshuaRogers24) [inline src/components/MembershipsList/index.js:71]: Do you think it is a little hard to see that ` m.Privacy = !m.Privacy` is the body of the if?

File: src/components/MembershipsList/index.js (hunk 1 above)
Comment (joshuaRogers24) [reply src/components/MembershipsList/index.js:31]: Actually, another question about this: does `involvement.Privacy` the same no matter the user? (Are there just some involvements that are set to private by default?) Or is `involvement.Privacy` different depending on the user?

File: src/components/MembershipsList/components/MembershipInfoCard/index.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+import { Link } from 'react-router-dom';
+import LockIcon from '@material-ui/icons/Lock';
+import useNetworkStatus from '../../../../hooks/useNetworkStatus';
+import './index.css';
+
+const MembershipInfoCard = ({ myProf, membership, onTogglePrivacy }) => {
+  const isOnline = useNetworkStatus();
+
+  const OnlineOnlyLink = ({ children }) => {
+    const linkClass = classnames({
```
Comment (joshuaRogers24) [inline src/components/MembershipsList/components/MembershipInfoCard/index.js:13]: This style class is a neat, helpful strategy!

File: src/components/MembershipsList/components/MembershipInfoCard/index.scss [hunk 5]
```diff
@@ -42,7 +54,7 @@
         border-left: 2px solid $primary-blue;
       }
     }
-    &-info-two {
+    &-picture {
       max-width: 100% !important;
```
Comment (joshuaRogers24) [inline src/components/MembershipsList/components/MembershipInfoCard/index.scss:58]: Is this classname used?

File: src/components/MembershipsList/index.js (hunk 3 above)
Comment (EjPlatzer) [reply src/components/MembershipsList/index.js:71]: So you're suggesting 
```suggestion
          if (m.MembershipID === membership.MembershipID) {
            m.Privacy = !m.Privacy;
          };
```
might be clearer? I personally find them equally clear but if you feel differently then I'm happy to change it.

Comment (EjPlatzer) [reply src/components/MembershipsList/index.js:71]: Actually after looking at it in context, I'm inclined to agree that brackets clarify the code. I'll update it.

**Review by joshuaRogers24 — APPROVED** [review]
Thanks for those changes. Looks good!

**Pull request merged**